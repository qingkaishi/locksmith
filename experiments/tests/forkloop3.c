#include <pthread.h>
#include <stdio.h>

void * dostuff(void * a) {
  *(int *)a = 0;
}

int main() {
  pthread_t t1;
  while(1) {
    int * mem = malloc(sizeof(int));
    *mem = 1;
    pthread_create(&t1, NULL, dostuff, mem);
    *mem = 1;
  }
}


