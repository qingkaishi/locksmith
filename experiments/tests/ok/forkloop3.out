************************* STARTING *************************

shared: alloc:forkloop3.c:11

Warning: Possible data race: alloc:forkloop3.c:11 is not protected!
 references:
  dereference of cast:forkloop3.c:5 at forkloop3.c:5
    alloc:forkloop3.c:11 => tmp::-1
                         => mem:forkloop3.c:11
                         => cast:forkloop3.c:5 at dostuff forkloop3.c:13
  locks acquired:
    <empty>
  in: main at forkloop3.c:8 

  dereference of mem:forkloop3.c:11 at forkloop3.c:12
    alloc:forkloop3.c:11 => tmp::-1
                         => mem:forkloop3.c:11
  locks acquired:
    <empty>
  in: main at forkloop3.c:8 

  dereference of mem:forkloop3.c:11 at forkloop3.c:14
    alloc:forkloop3.c:11 => tmp::-1
                         => mem:forkloop3.c:11
  locks acquired:
    <empty>
  in: main at forkloop3.c:8 


*************************** DONE ***************************
